src/lib/mongoBackend/location.cpp

```diff
@@ -319,37 +321,6 @@ static bool getGeoJson
   std::vector<double>      coordLong;
   orion::BSONArrayBuilder  ba;
 
-  if ((apiVersion == V1) && (caP->type != GEO_POINT) && (caP->type != GEO_LINE) && (caP->type != GEO_BOX) &&
-      (caP->type != GEO_POLYGON) && (caP->type != GEO_JSON))
-  {
-    // This corresponds to the legacy way in NGSIv1 based in metadata
-    // The block is the same that for GEO_POINT but it is clearer if we keep it separated
-
-    __sync_fetch_and_add(&noOfDprLocationMetadata, 1);
-    if (logDeprecate)
-    {
-      LM_W(("Deprecated usage of metadata location %s detected in attribute %s at entity update, please use geo:json instead", caP->type.c_str(), caP->name.c_str()));
-    }
-
-    double  aLat;
-    double  aLong;
-
-    if (!string2coords(caP->stringValue, aLat, aLong))
-    {
-      *errDetail = "geo coordinates format error [see Orion user manual]: " + caP->stringValue;
-      return false;
-    }
-
-    geoJson->append("type", "Point");
-
-    orion::BSONArrayBuilder ba;
-    ba.append(aLong);
-    ba.append(aLat);
-    geoJson->append("coordinates", ba.arr());
-
-    return true;
-  }
-
```

Comment:
This seems to be a big block of duplicated code. The changes done PR https://github.com/telefonicaid/fiware-orion/pull/4491 confirm and fix it.

NTC (post-merge comment)